### ISSUE TITLE
camera-preview is not working.

### ISSUE THREAD

**Issue by louisren** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#247]
Hi All,

The camera-preview component not working in my project when choose ion-fab click event. my source code as below:
1.
![2017-03-28_12-06-08](https://cloud.githubusercontent.com/assets/4684527/24388826/d06558a2-13af-11e7-8737-2d303f019f2f.png)

2. 
![2017-03-28_12-11-23](https://cloud.githubusercontent.com/assets/4684527/24388833/d9ec3c9c-13af-11e7-8e95-217e176f94a7.png)

import {Component} from '@angular/core';
import {NavController, NavParams} from 'ionic-angular';
import {CameraPreview, CameraPreviewOptions} from "@ionic-native/camera-preview";

/*
 Generated class for the CameraPreview page.

 See http://ionicframework.com/docs/v2/components/#navigation for more info on
 Ionic pages and navigation.
 */
@Component({
  selector: 'page-camera-preview',
  templateUrl: 'camera-preview.html'
})
export class CameraPreviewPage {

  constructor(public navCtrl: NavController, public navParams: NavParams, private cameraPreview: CameraPreview) {
    console.info("=========" + cameraPreview);
  }

  ionViewDidLoad() {
    console.log('ionViewDidLoad CameraPreviewPage');
  }

  takePicture() {
    const cameraPreviewOpts: CameraPreviewOptions = {
      x: 40,
      y: 40,
      width: window.screen.width,
      height: window.screen.height,
      camera: 'rear',
      tapPhoto: true,
      previewDrag: true,
      toBack: true,
      alpha: 1
    };

    // start camera
    this.cameraPreview.startCamera(cameraPreviewOpts).then(
      (res) => {
        console.log(res)
      },
      (err) => {
        console.log(err)
      });

  }
}

**Comment by rrundle** [comment]
I'm interested in the feedback here because I have the same setup and the startCamera success callback returns <em>null</em>. Although I do not have the <em>ion-fab</em>element.

**Comment by louisren** [comment]
Would you please tell me how to resolve this issue.

**Comment by louisren** [comment]
I have resolved this issue, you can refer to the following screenshot. You need to set the "toBack" property is false. 
![2017-03-29_10-25-12](https://cloud.githubusercontent.com/assets/4684527/24435802/19c484c2-146a-11e7-9b54-52a0b5088697.png)

**Comment by rrundle** [comment]
Yes I has success doing the same, but it should work if that is set to true, with the html elements laid over it as long as the body, html, ion-app, and ion-content are set to background-color: transparent. Not having success with that (although that was not your issue).

**Comment by FrancisTse8** [comment]
Hello @louisren,

I have the same problem not being able to show the camera preview.  I have tried setting toBack to 'true' and 'false'.  Neither one worked.  I noticed that your plugin and mine looked different.  Don't know if that is the issue.
Below are screen shots of the plugin for my project:
![plugin_image](https://cloud.githubusercontent.com/assets/15918099/24732973/75913b28-1a42-11e7-9343-6168b903b44c.PNG)
Here is the package.json file showing the name and version of the plugin:
![json_image](https://cloud.githubusercontent.com/assets/15918099/24733101/7e6edfa6-1a43-11e7-9a62-2f0c10b25829.PNG)

After I created the ionic2 project, I did the following to add the plugin per the ionic-native documentation:
>ionic plugin add cordova-plugin-camera-preview
>npm install --save @ionic-native/camera-preview

BTW, I had to add CameraPreview to the import and providers of the app.module.ts to get rid of some errors.  Did you have to do that also?

Do you see any issue with what I did that might keep camera-preview from working?

Thanks for any help you can provide.

**Comment by louisren** [comment]
You can refer to the following camera preview component.
cordova plugin add https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview.git
npm install --save @ionic-native/camera-preview
![2017-04-06_10-19-00](https://cloud.githubusercontent.com/assets/4684527/24734880/853e160c-1ab2-11e7-8ccb-aaa2b0e9c7e7.png)

**Comment by FrancisTse8** [comment]
It appears that my camera-preview version is 3.4.4.  See below:
![package_json](https://cloud.githubusercontent.com/assets/15918099/24760052/7c393b6a-1ab5-11e7-9f81-c9dec5ac9800.PNG)

I did the command 'npm install --save @ionic-native/camera-preview' originally.  Is there a way I can specify which version of camera-preview to install?  I tried changing my package.json file's "dependencies" to "@ionic-native/camera-preview": "^3.3.0" and ran the app again but still no preview.

**Comment by FrancisTse8** [comment]
Happy to report that I now have camera-preview showing the camera image.  Thanks @louisren for point me to the right direction!!!  I had to google how to remove and reinstall the camera-preview plugin.  For future reference and others who might find this useful, I will document what I did.

First, when I did a 'ionic plugin list', I got the following:
com.mbppower.camerapreview 0.0.8 "cordova-plugin-camera-preview"
cordova-plugin-camera-preview 0.9.0 "cordova-plugin-camera-preview"
cordova-plugin-console 1.0.5 "Console"
cordova-plugin-device 1.1.4 "Device"
cordova-plugin-splashscreen 4.0.2 "Splashscreen"
cordova-plugin-statusbar 2.2.1 "StatusBar"
cordova-plugin-whitelist 1.3.1 "Whitelist"
ionic-plugin-keyboard 2.2.1 "Keyboard"
Two camera-preview plugins - don't know how I ended up with the two.

I tried removing the com.mbppower.camerapreview plugin with the command 'ionic plugin rm com.mbppower.camerapreview' first.  Ran the app and camera-preview still did not work.  I ended up removing the second camera-preview plugin and did the 'cordova plugin add https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview.git'.

When I do the 'ionic run android', camera-preview worked.  BTW, I forgot to do the 'npm install --save @ionic-native/camera-preview' command.  However, when I looked at the @ionic-native folder, there was a camera-preview folder there with yesterday's date.  So, it looks like the 'ionic plugin rm ~' command did not remove the camera-preview folder and the content in that folder is good.

Just to make sure, I did the 'npm install --save @ionic-native/camera-preview' command and a new folder was created with the current.  When I ran my app, camer-preview still works!

In summary, one has to be sure that the correct plugin has been installed.  Check things by doing 'ionic plugin list'.  If in doubt, do the 'ionic plugin rm ~' and reinstall.

**Comment by FrancisTse8** [comment]
Now that I've got camera-preview to work, I think I am coming to the same issue that @rrundle had.  For some reason, I am not able to show the camera-preview with toBack: set to 'true' even when all the HTML objects where the camera-preview area that is shown is set to transparent.  Below are the simple HTML and CSS code that I am testing together with the code to set up the camera-preview.  I also tried playing with setting <body> and <html> in index.html to have background-color: transparent.  That did not work either.  Didn't know where to set background-color for <ion-app>.  All works well when toBack: is set to 'false'.  Any ideas what else to try?  @rrundle, were you able to get camera-preview to work with toBack: set to 'true'?
![html](https://cloud.githubusercontent.com/assets/15918099/24774532/4d615e30-1ae7-11e7-94d9-d3c061f5360c.PNG)
![css](https://cloud.githubusercontent.com/assets/15918099/24774536/5119a500-1ae7-11e7-9920-7fa037cbbf1a.PNG)
![ts](https://cloud.githubusercontent.com/assets/15918099/24774541/56e0bece-1ae7-11e7-9fd8-50854090e10a.PNG)

**Comment by asadwaheed1** [comment]
@rrundle 

> Yes I has success doing the same, but it should work if that is set to true, with the html elements laid over it as long as the body, html, ion-app, and ion-content are set to background-color: transparent. Not having success with that (although that was not your issue).


How can i set background transparent its not happening in android build. It does not show camera preview  when toBack is true.

**Comment by rrundle** [comment]
I think you need to look elsewhere to see if you have everything else declared properly. I have toBack set to true and it builds properly in android. Are you passing the options to the method with CameraPreviewOptions?

  import { CameraPreview, CameraPreviewOptions } from '@ionic-native/camera-preview';

  private options: CameraPreviewOptions;
  private cameraPreview: CameraPreview,

Then in your code just define your options object and pass it to the startCamera method.
 
  this.options = {
        x: 0,
        y: 0,
        width: window.screen.width,
        height: window.screen.height,
        camera: 'front',
        toBack: true,
        tapPhoto: false,
        previewDrag: false
      };

  this.cameraPreview.startCamera(this.options)
    .then(response => {
       console.log('camera running!' + response)
    })
    .catch(error => {
       console.log('couldnt access camera' + error)
    })

This is my css in app.scss:
    html, body, ion-app, ion-content, ion-page, .nav-decor {
      background-color: transparent !important;
    }

If you are getting a build error what does the error tell you? If you are using Ionic be sure you installed the plugin and it is displaying in your config.xml and that you have installed the ionic module from npm. 

config.xml:
    <plugin name="cordova-plugin-camera-preview" spec="https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview.git" />

package.json:
        "@ionic-native/camera-preview": "^3.4.4",

**Comment by asadwaheed1** [comment]
I did the above but i am using a navBar and then cameraPage. but i am using this plugin "cordova-plugin-camera-preview": "git+https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview.git"

**Comment by asadwaheed1** [comment]
in cordova plugin list its showing cordova-plugin-camera-preview 0.9.0 "cordova-plugin-camera-preview"

**Comment by asadwaheed1** [comment]
And I am not getting any build error. just white background and camera preview not showing. When i press start camera it blinks like a screen opens and closes but no errors

**Comment by asadwaheed1** [comment]
@rrundle I am sorry for complaining. It worked!

**Comment by rrundle** [comment]
👍 Sometimes you have to uninstall and reinstall the platform, but glad it worked out.

**Comment by apuravchauhan** [comment]
If someone ran into this issue using Ionic 4, I had to add the following bare minimum in my page's scss file

page-home {
  ion-content {
    background-color: transparent !important;
  }
}
 ion-app,.nav-decor {
  background-color: transparent !important;
}

**Comment by kathan3009** [comment]
i am getting this error: {camera preview is not working}{PLZ help}

 PracticePage.html:53 ERROR TypeError: Object(...) is not a function
    at CameraPreview.startCamera (index.js:66)
    at PracticePage.webpackJsonp.428.PracticePage.takePicture (practice.ts:48)
    at Object.eval [as handleEvent] (PracticePage.html:53)
    at handleEvent (core.js:13589)
    at callWithDebugContext (core.js:15098)
    at Object.debugHandleEvent [as handleEvent] (core.js:14685)
    at dispatchEvent (core.js:10004)
    at core.js:10629
    at HTMLButtonElement.<anonymous> (platform-browser.js:2628)
    at t.invokeTask (polyfills.js:3)

**Comment by Anusha14-cell** [comment]
> I'm interested in the feedback here because I have the same setup and the startCamera success callback returns _null_ (so camera does not turn on). Although I do not have the _ion-fab_ element.

same problem here.
